plasmapy/formulary/mathematics.py

```diff
@@ -94,73 +91,6 @@ def Fermi_integral(
         raise TypeError(f"Improper type {type(x)} given for argument x.")
 
 
-@vectorize(
-    [
-        float64(float64),
-    ]
-)
```

Comment:
I was digging into this `_UFunNoLoopError`, which is causing all the test failures including the `hypothesis` tests, and I'm almost 100% certain it's caused by `numba.vectorize` and it's incompatibility with the current development branch of `numpy`.  From what I can tell, `numba.vectorize` is creating a `ufunc` signature that looks like `float64 -> None` while `numpy` requires `float64 -> float64` and, thus, `numpy` is complaining.

So, I'll leave the solution up to you, but you can do either...

1. what you're doing here and delteing `Chandrasekhar_G`
2. remove `numpy-dev` tests and set dependencies to `numpy < 1.22`

Personally, I don't like deleting functionality that "isn't broken" so I'd go with placing the dependency restriction, unless we really want what is in `numpy==1.22`.